src/modules/dashboard/components/ColonyMembers/MemberControls/MemberControls.tsx

```diff
@@ -0,0 +1,181 @@
+import WrongNetworkDialog from '~dialogs/WrongNetworkDialog';
+import ManageWhitelistDialog from '~dashboard/Dialogs/ManageWhitelistDialog';
+
+import { Colony, useColonyExtensionsQuery, useLoggedInUser } from '~data/index';
+import { useTransformer } from '~utils/hooks';
+import { useEnabledExtensions } from '~utils/hooks/useEnabledExtensions';
+import { checkIfNetworkIsAllowed } from '~utils/networks';
+import { getAllUserRoles } from '~modules/transformers';
+import { hasRoot, canAdminister, canArchitect } from '~modules/users/checks';
+import { oneTxMustBeUpgraded } from '~modules/dashboard/checks';
+
+const displayName = 'dashboard.MemberControls';
+
+const MSG = defineMessages({
+  editPermissions: {
+    id: 'dashboard.MemberControls.editPermissions',
+    defaultMessage: 'Edit permissions',
+  },
+  banAddress: {
+    id: 'dashboard.MemberControls.banAddress',
+    defaultMessage: 'Ban address',
+  },
+  unbanAddress: {
+    id: 'dashboard.MemberControls.unbanAddress',
+    defaultMessage: 'Unban address',
+  },
+  manageWhitelist: {
+    id: 'dashboard.MemberControls.manageWhitelist',
+    defaultMessage: 'Manage address book',
+  },
+});
+
+interface Props {
+  colony: Colony;
+}
+
+const MemberControls = ({
+  colony,
+  colony: { colonyAddress, version, isDeploymentFinished },
+}: Props) => {
+  const {
+    networkId,
+    username,
+    ethereal,
+    walletAddress: currentUserWalletAddress,
+  } = useLoggedInUser();
+  const isNetworkAllowed = checkIfNetworkIsAllowed(networkId);
+  const hasRegisteredProfile = !!username && !ethereal;
+
+  const openWrongNetworkDialog = useDialog(WrongNetworkDialog);
+  const openToggleBanningDialog = useDialog(BanUserDialog);
+
+  const { isVotingExtensionEnabled } = useEnabledExtensions({
+    colonyAddress,
+  });
+
+  useEffect(() => {
+    if (!ethereal && !isNetworkAllowed) {
+      openWrongNetworkDialog();
+    }
+  }, [ethereal, isNetworkAllowed, openWrongNetworkDialog]);
+
+  const { data: colonyExtensions } = useColonyExtensionsQuery({
+    variables: { address: colonyAddress },
+  });
+
+  const openPermissionManagementDialog = useDialog(PermissionManagementDialog);
+
+  const handlePermissionManagementDialog = useCallback(() => {
+    openPermissionManagementDialog({
+      colony,
+      isVotingExtensionEnabled,
+    });
+  }, [openPermissionManagementDialog, colony, isVotingExtensionEnabled]);
+
+  const openToggleManageWhitelistDialog = useDialog(ManageWhitelistDialog);
+
+  const handleToggleWhitelistDialog = useCallback(() => {
+    return openToggleManageWhitelistDialog({
+      colony,
+    });
+  }, [openToggleManageWhitelistDialog, colony]);
+
+  // eslint-disable-next-line max-len
+  const oneTxPaymentExtension = colonyExtensions?.processedColony?.installedExtensions.find(
+    ({ details, extensionId: extensionName }) =>
+      details?.initialized &&
+      !details?.missingPermissions.length &&
+      extensionName === Extension.OneTxPayment,
+  );
```

Comment:
We keep repeating this code block over and over _(through the Dapp)_ in order to detect an installed extension.

We need a better system for this. 

I don't have a good answer of the top of my head, but definitely something to think about

Reply:
I made an issue to remember about it.